src/NServiceBus.Core/OpenTelemetry/Metrics/MessagingMetricsMeters.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace NServiceBus;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Pipeline;
+
+class MessagingMetricsMeters
+{
+    public MessagingMetricsMeters(IMeterFactory meterFactory)
+    {
+        // TODO the problem with this approach is that it's harder to do approval testing on the exposed meters
+        // TODO we also probably need to keep the meter around to assert that the version did not change 
+        var meter = meterFactory.Create("NServiceBus.Core", "0.2.0");
+        totalProcessedSuccessfully = meter.CreateCounter<long>(Metrics.TotalProcessedSuccessfully,
+            description: "Total number of messages processed successfully by the endpoint.");
+        totalFetched = meter.CreateCounter<long>(Metrics.TotalFetched,
+            description: "Total number of messages fetched from the queue by the endpoint.");
+        totalFailures = meter.CreateCounter<long>(Metrics.TotalFailures,
+            description: "Total number of messages processed unsuccessfully by the endpoint.");
+        messageHandlerTime = meter.CreateHistogram<double>(Metrics.MessageHandlerTime, "s",
+            "The time in seconds for the execution of the business code.");
+        criticalTime = meter.CreateHistogram<double>(Metrics.CriticalTime, "s",
+            "The time in seconds between when the message was sent until processed by the endpoint.");
+    }
+
+    public void RecordMessageSuccessfullyProcessed(TagList tags) => totalProcessedSuccessfully.Add(1, tags);
```

Comment:
Another thing to consider in favor of this approach is the best practice in the document @danielmarbach shared:
> [System.Diagnostics.Metrics.Meter](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter) implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meter objects it creates, disposing them when the DI container is disposed. It's unnecessary to add extra code to invoke Dispose() on the Meter, and it won't have any effect.[System.Diagnostics.Metrics.Meter](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter) implements IDisposable, but IMeterFactory automatically manages the lifetime of any Meter objects it creates, disposing them when the DI container is disposed. It's unnecessary to add extra code to invoke Dispose() on the Meter, and it won't have any effect.

Reply:
> Another problem with this approach is that the amount of methods here will continue to increase and they will become less correlated, which then makes this approach require the feature per metric approach, but we also discussed some of the downsides of that approach in the [other PR yesterday](https://github.com/Particular/NServiceBus/pull/7082).

I would love to hear those arguments because what I have seen so far was not really organizing metrics by feature.

Reply:
@mauroservienti documented them in the linked PR description

Reply:
I think this is the wrong PR for this discussion, isn't it?

Reply:
Moved things over